### PR TITLE
Streaming write support for BMFF

### PR DESCRIPTION
**Pull request by mauricefisher64**
## Changes in this pull request
Adds streaming write support for BMFF.  Support XMP in BMFF formats.

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [x] All applicable changes have been documented.
- [x] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

**Review by gpeacock — COMMENTED** [review]
Thank you so much for doing this, it looks great!

File: sdk/src/asset_handlers/bmff_io.rs [hunk 1]
```diff
@@ -251,7 +257,7 @@ pub(crate) struct BoxInfoLite {
     pub size: u64,
 }
 
-fn read_box_header_ext(reader: &mut dyn CAIRead) -> Result<(u8, u32)> {
+fn read_box_header_ext<R: Read + Seek + ?Sized>(reader: &mut R) -> Result<(u8, u32)> {
```
Comment (gpeacock) [inline sdk/src/asset_handlers/bmff_io.rs:260]: if only we could do this everywhere!

**Review by gpeacock — APPROVED** [review]
Have a great sabbatical!

File: sdk/src/asset_handlers/bmff_io.rs (hunk 1 above)
Comment (mauricefisher64) [reply sdk/src/asset_handlers/bmff_io.rs:260]: It may be worth trying again.  I think the issue I had was was with Sync and Send.

**Pull request merged**